enterprise/backend/src/metabase_enterprise/serialization/upsert.clj

```diff
@@ -53,14 +53,16 @@
 ;; This could potentially be unrolled into one giant select
 (defn- select-identical
   [model entity]
-  (->> (or (identity-condition model)
-           (throw (ex-info (trs "Model {0} does not support upsert" model) {:model model})))
-       (select-keys entity)
-       (m/map-vals (fn [v]
-                     (if (coll? v)
-                       (json/encode v)
-                       v)))
-       (m/mapply t2/select-one model)))
+  (let [entity-id (if (:entity_id entity) [:entity_id] [])]
+    (->> (or (identity-condition model)
+             (throw (ex-info (trs "Model {0} does not support upsert" model) {:model model})))
+         (into entity-id)
```

Comment:
If I'm understanding this correctly, this will mean we need to clear the destination's `entity_id`s first. Otherwise a user with `entity_id` mismatches will get duplicates.

Reply:
We shouldn't check entity-id for identity-matching, right?
The purpose here is to overwrite entity-ids based on Serdes v1 identity-condition

Reply:
Hmmm but this is not identity matching, this is "has data changed" call, if I understand correctly. I mean test passes right now and does not generate duplicates even though entity ids are different?

Reply:
@bshepherdson I have thought about this a little bit more and I agree this change looks like it could lead to duplicates. IDK why it did not in my test. But now I've replaced it with just `--mode update` command option and it seems like this is what we need?